src/meta/src/stream/fragmenter.rs

```diff
@@ -124,7 +124,30 @@ where
             .generate_interval::<{ IdCategory::Actor }>(actor_len as i32)
             .await? as _;
 
-        let stream_graph = self.stream_graph.build(ctx, start_actor_id, actor_len)?;
+        // Compute how many table ids should be allocated for all actors.
+        let mut table_ids_cnt = 0;
+        for (local_fragment_id, fragment) in self.fragment_graph.fragments() {
+            let num_of_actors = self
+                .fragment_actors
+                .get(local_fragment_id)
+                .expect("Fragment should have at least one actor")
+                .len();
+            // Total table ids number equals to table ids cnt in all actors.
+            table_ids_cnt += num_of_actors * fragment.table_ids_cnt;
```

Comment:
I believe it's up to the frontend to tell how many table ids an operator needs and whether it requires a shared state among executors.

Reply:
Good point. I'll think a bit whether easy to do this.

Reply:
Actually no. In frontend even if we can traverse the plan node tree, we do not know the actors number for each fragment. And there's even no 'fragment' concept in frontend. But yes, for some static info, like hash join table ids = 2, agg table ids = 1, I'm not sure how to send these from frontend. Note that different actors should have different table ids, just like the opeartor id. 

> whether it requires a shared state among executors.
I think the operators set using relational states are fixed? This sounds like a dynamic flag to control whether use relational states.

Reply:
> But yes, for some static info, like hash join table ids = 2, agg table ids = 1

Yep, that's what I mean. As for the parallelism and the real number of table ids to allocate, we can calculate on the meta service based on these info provided by the frontend. I just prefer not to recognize different kinds of nodes on meta.

Reply:
Good to consider when we add more stateful executors.